### PR TITLE
Feature/drone state mapper

### PR DESCRIPTION
**Pull request by Segelzwerg**

### CODE REVIEW

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/input_layer/drone_state_mapper.py  2
- src/drone_controller/input_layer/drone_state.py  2
- src/drone_controller/exception/exceptions.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5277357&bid=17281593)

File: src/drone_controller/input_layer/drone_state_mapper.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        return [a * b for a, b in zip(first_list, second_list)]
+
+    @staticmethod
+    def normalize_rotation(rotation) -> list:
+        """
+        Converts the drone rotation to values in [-1,1]
+        :param rotation: 3D with angle values
+        :return: 3D vector
+        """
+        normalized_rotations = list([math.sin(x) for x in rotation[0:2]])
```
Comment (TreeKid) [inline src/drone_controller/input_layer/drone_state_mapper.py]: convert x to radians before sin()

File: src/drone_controller/input_layer/drone_state_mapper.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        :param user_input: keyboard input from user
+        :return: the expected state of the drone
+        """
+        state: dict = current_state.state_dict
+        rotation_normalized = self.normalize_rotation(state['Rotation'])
+
+        acceleration = [i * user_input['Acceleration'] for i in rotation_normalized]
+        state['Acceleration'] = self.add_lists(state['Acceleration'], acceleration)
+
+        state['Velocity'] = self.add_lists(state['Velocity'], state['Acceleration'])
```
Comment (TreeKid) [inline src/drone_controller/input_layer/drone_state_mapper.py]: Make a command to explain why you can add m/s to m/s²

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/input_layer/drone_state_mapper.py  3
- src/drone_controller/input_layer/drone_state.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5277357&bid=17281593)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/input_layer/drone_state_mapper.py  3
- src/drone_controller/input_layer/drone_state.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5277357&bid=17281593)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/input_layer/drone_state_mapper.py  3
- src/drone_controller/input_layer/drone_state.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5277357&bid=17281593)

File: test/drone_controller/input_layer/test_drone_state_mapper.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        rotation = ZERO_VECTOR
+        velocity = ZERO_VECTOR
+        velocity_ang = ZERO_VECTOR
+        acceleration = ZERO_VECTOR
+        acceleration_ang = ZERO_VECTOR
+
+        drone_state = DroneState(position, rotation, velocity, velocity_ang, acceleration,
+                                 acceleration_ang)
+
+        position = [0.174, 0.0, 0.985]
```
Comment (TreeKid) [inline test/drone_controller/input_layer/test_drone_state_mapper.py:101]: should be y axis

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/input_layer/drone_state_mapper.py  3
- src/drone_controller/input_layer/drone_state.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5277357&bid=17281593)

**Comment by Segelzwerg** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/drone_controller/input_layer/drone_state_mapper.py  3
- src/drone_controller/input_layer/drone_state.py  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/treeandsea/DroneController/pullRequest?prid=5277357&bid=17281593)
